### PR TITLE
Update `end_slot_hook`

### PR DESCRIPTION
**Pull request by bkolad**
# Description
This PR adds `state_root` to the `end_slot_hook` (which the EVM module requires). We do it as a separate PR because it influences the #615 discussion. 

## Linked Issues
- Related to #737 

## Testing
Ci passes.

## Docs
No update.

### CODE REVIEW

File: module-system/sov-modules-stf-template/src/lib.rs [hunk 1]
```diff
@@ -99,8 +99,7 @@ where
             .expect("jellyfish merkle tree update must succeed");
 
         let mut working_set = WorkingSet::new(self.current_storage.clone());
-
-        self.runtime.end_slot_hook(&mut working_set);
+        self.runtime.end_slot_hook(root_hash, &mut working_set);
```
Comment (bkolad) [inline module-system/sov-modules-stf-template/src/lib.rs:102]: @neysofu we will need to update the `non-JMT` data in the `end_slot_hook` but with the latest state root hash. This raises the question of when we save the `non-JMT` data.

Comment (neysofu) [reply module-system/sov-modules-stf-template/src/lib.rs:102]: Yup, as agreed during our call. I think we have a plan to tackle this.

File: examples/demo-stf/src/runtime.rs [hunk 2]
```diff
@@ -119,8 +119,13 @@ impl<C: Context, Cond: ValidityCondition> SlotHooks<Cond> for Runtime<C> {
 
     fn end_slot_hook(
         &self,
-        _working_set: &mut sov_state::WorkingSet<<Self::Context as sov_modules_api::Spec>::Storage>,
+        #[allow(unused_variables)] root_hash: [u8; 32],
```
Comment (neysofu) [inline examples/demo-stf/src/runtime.rs:122]: I didn't know you could add argument-specific attributes. Nice!

**Review by neysofu — APPROVED** [review]
LGTM

**Pull request merged**